### ISSUE TITLE
Merged doctest compilation time is not reported if all doctests are merged

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust#146960]
I tested this code:

```rust
/// ```
/// playground::foo();
/// ```
pub fn foo() {}
```

I expected to see this happen: As per closed issue #144270 and PR #144308, the time spent compiling the doctests should be reported.

Instead, this happened: The compilation time is not reported (all visible timestamps are `0.00 s`), *unless* at least one doctest in the crate is compiled separately for any reason (e.g. <code>\`\`\`compile_fail</code>).

### Meta

`rustc --version --verbose`:
```
rustc 1.92.0-nightly (4645a7988 2025-09-17)
binary: rustc
commit-hash: 4645a7988177c286f61609cc667ecae4c571a2e8
commit-date: 2025-09-17
host: aarch64-apple-darwin
release: 1.92.0-nightly
LLVM version: 21.1.1
```

@rustbot label A-doctests A-libtest
cc @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
Opened https://github.com/rust-lang/rust/pull/147032.
